Makefile

```diff
@@ -0,0 +1,44 @@
+# This Makefile just indirects to call gradlew
+
+SPARK_VERSION :=2.2.0
+SPARK_HOME    :=/usr/local/java/spark-${SPARK_VERSION}-bin-hadoop2.7
+
+CXX  := /usr/bin/g++ -std=c++14 -Wno-deprecated
+FLEX := /usr/bin/flex
+LIBFLEX := -ll
+
+.PHONY: shadowJar go test
+
+all: shadowJar go
+
+shadowJar:
+	./gradlew -Dspark.version=${SPARK_VERSION} shadowJar archiveZip
+
+nativeLibDebug:
+	./gradlew -Dspark.version=${SPARK_VERSION} nativeLibDebug
+
+nativeLibClean:
+	./gradlew -Dspark.version=${SPARK_VERSION} nativeLibClean
+
+go:
+	./gradlew -Dspark.version=${SPARK_VERSION} -Dspark.home=${SPARK_HOME} \
+	  -Dtest.single=AnnotationsSuite test
```

Comment:
this seems like something you might have left in?

Reply:
This top-level Makefile is for my convenience, because I have no interest in typing a
long gradlew command when I could say "make" or "make test".  If anyone else ever
uses it at all, I expect they would change the "go" target to suit whatever they're
working on in their own development branches.

Up to now I've kept the Makefile only in my own branches.  But that ends up being a pain
because if it isn't in every branch, then git complains about it when switching between different
branches, some of which have it and others which don't.

Reply:
I sympathize with gradle woes. Is the issue that you have a spark version other than the default 2.2.0? Perhaps gradle should read the spark version from an environment variable. WRT `SPARK_HOME`, I believe other hailers sets that in their `.profile`s. with those to set it's just `gradle test` or `gradle testPython`.

WRT, checking in this file or not, if you don't version control it at all, then it seems you wouldn't have trouble with git when changing branches? Is anything in here specific to a branch?

Reply:
> my convenience

If we all add stuff for our convenience, which is to say, not standard workflows but our personal workflow (which `-Dtest.single=AnnotationsSuite` clearly is), then the codebase will become a mess.

This is what e.g. local aliases are for:

```
$ alias gw
alias gw='./gradlew'
```

Then there's no conflict or pollution.  And `gw` is even shorter than `make`.

 > I expect they would change the "go" target to suit whatever they're working on in their own development branches

Then everyone is going to have the same problem you did with the Makefile: the changes will be different in different branches, git will complain, and they will generate merge conflicts.